src/syscheckd/create_db.c

```diff
@@ -1574,6 +1591,83 @@ void fim_diff_folder_size(){
     os_free(diff_local);
 }
 
+void update_wildcards_config(OSList *directories,
+                             OSList *wildcards){
```

Comment:
This function mixes a global mutex (`syscheck.directories_lock`) with local parameters (`directories` and `wildcards`). Either move the lock outside of the function and leave it as pure as possible or remove the parameters and use the global pointers directly.